discovery/discovery_test.go

```diff
@@ -727,15 +716,7 @@ func TestCachedPool_GetOrchestrators_MaxBroadcastPriceNotSet(t *testing.T) {
 	server.BroadcastCfg.SetMaxPrice(nil)
 	gmp := runtime.GOMAXPROCS(50)
 	defer runtime.GOMAXPROCS(gmp)
-	var mu sync.Mutex
-	first := true
 	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL, params server.GetOrchestratorInfoParams) (*net.OrchestratorInfo, error) {
-		mu.Lock()
-		if first {
```

Comment:
why were we pausing on the first request, don't really understand that

Reply:
me neither - best guess is some early test needed it or had a race condition (probably the deadlock one), and it got copied everywhere else.

I double checked the rest of these tests and seems that I mistakenly removed the sleep from the deadlock test; that apparently was meant to drop one of the orchs so it wouldn't pass discovery. Re-added it, hopefully with a little better explanation in a6110ecaa5840d0c1e78b3c4cd26ea52da076aa1 (I still have no idea how this could have caused a deadlock though.)